new-log-viewer/src/services/decoders/JsonlDecoder/index.ts

```diff
@@ -0,0 +1,242 @@
+            const logEvent = this.#logEvents[logEventIdx] as JsonLogEvent;
+            logLevel = logEvent.level;
+            message = this.#formatter.formatLogEvent(logEvent);
+            timestamp = logEvent.timestamp.valueOf();
+        }
+
+        return [
+            message,
+            timestamp,
+            logLevel,
+            logEventIdx + 1,
+        ];
+    };
+
+    /**
+     * Parses each line from the data array as a JSON object and buffers it internally. If a
+     * line cannot be parsed as a JSON object, an error is logged and the line added to an
+     * invalid map.
+     *
+     * NOTE: The data array(file) is freed after the very first run of this method.
+     */
+    #deserialize () {
+        if (null === this.#dataArray) {
+            return;
+        }
+
+        const text = JsonlDecoder.#textDecoder.decode(this.#dataArray);
+        let beginIdx: number = 0;
+        while (beginIdx < text.length) {
+            const endIdx = text.indexOf("\n", beginIdx);
+            const line = (-1 === endIdx) ?
+                text.substring(beginIdx) :
+                text.substring(beginIdx, endIdx);
+
+            beginIdx = (-1 === endIdx) ?
+                text.length :
+                endIdx + 1;
+
+            this.#parseJson(line);
+        }
+
+        this.#dataArray = null;
+    }
+
+    /**
+     * Parse line into a json log event and buffer internally. If the line contains invalid
+     * json, an entry is added to invalid log event map.
+     *
+     * @param line
+     */
+    #parseJson (line: string) {
+        try {
+            const fields = JSON.parse(line) as JsonValue;
+            if (!isJsonObject(fields)) {
+                throw new Error("Unexpected non-object.");
+            }
+            this.#logEvents.push({
+                fields: fields,
+                level: convertToLogLevelValue(fields[this.#logLevelKey]),
+                timestamp: convertToDayjsTimestamp(fields[this.#timestampKey]),
+            });
+        } catch (e) {
+            if (0 === line.length) {
+                return;
+            }
+            console.error(e, line);
+            const currentLogEventIdx = this.#logEvents.length;
+            this.#invalidLogEventIdxToRawLine.set(currentLogEventIdx, line);
+            this.#logEvents.push({
+                fields: {},
+                level: LOG_LEVEL.NONE,
+                timestamp: convertToDayjsTimestamp(INVALID_TIMESTAMP_VALUE),
+            });
+        }
+    }
+
+    /**
+     * Filters log events and generates an internal array which serves as a mapping between filtered
+     * events and the original log event index. Sets the map to null if the filter is null.
+     *
+     * @param logLevelFilter
+     */
+    #filterLogs (logLevelFilter: LogLevelFilter) {
+        this.#filteredLogEventMap = null;
+
+        if (null === logLevelFilter) {
+            return;
```

Comment:
```suggestion
        if (null === logLevelFilter) {
            this.#filteredLogEventMap = null;
            return;
```

I agree with the Rabbit on this one.